src/functions.php

```diff
@@ -38,3 +40,36 @@ function usleep(int $time, LoopInterface $loop) : PromiseInterface
 {
     return Timer\resolve($time/1000000, $loop);
 }
+
+/**
+ * Returns the content type in MIME format, like text/plain or application/octet-stream, or FALSE on failure.
+ *
+ * @param string $fileName
+ * @param LoopInterface $loop
+ *
+ * @return PromiseInterface
+ */
+function mime_content_type(string $fileName, LoopInterface $loop) : PromiseInterface
+{
+    $process = new Process(sprintf('php -r "echo mime_content_type(\'%s\');"', $fileName));
+    $process->start($loop);
+    $deferred = new Deferred();
+    $data = '';
+    $stdout = $process->stdout;
+
+    $stdout->on('data', function($chunk) use (&$data) {
+        $data .= $chunk;
+    });
+
+    $stdout->on('end', function() use (&$data, $deferred, $fileName) {
+        empty($data === '')
```

Comment:
I think to according to PHP Docs, we should check empty and warning messages. If you have warnings enabled the output will contain warnings. Something like this:

```php
$isFailure = strpos($data, 'Warning') !== false || empty($data);
```

Reply:
Even better. We can disable error_reporting in this specific child process. The documentation says that only warnings are thrown, and we know how to manage them.

Reply:
Yes, it will be better 🔥